accel/tcg/cputlb.c

```diff
@@ -1836,6 +1855,20 @@ uint64_t cpu_ldq_data_ra(CPUArchState *env, target_ulong ptr, uintptr_t retaddr)
     return cpu_ldq_mmuidx_ra(env, ptr, cpu_mmu_index(env, false), retaddr);
 }
 
+#ifdef TARGET_CHERI
+/*
+ * TODO(am2419): Ugly hack to avoid logging memory accesses that load capability
+ * components as normal memory accesses. The caller is responsible for logging.
+ */
```

Comment:
I think logging the two memory reads is fine since it shows the PESBT and cursor values. Might be useful to debug compression issues instead of logging the two values separately afterwards.

That would also allow moving `log_instr_load_int` into cpu_load_helper.
If we want to avoid duplicate logging, I think having a MO_CAP is nicer long-term.

Reply:
This was a point of contention for me. The reason I opted for this approach is that I want to distinguish explicitly that the value being loaded/stored are capability bytes.
I agree that ultimately we should treat capabilities as a new data type and handle both the tag and data within TCG ops, so having the two load/store separated breaks this idea of having `tcg_gen_cap_*` that operate on an hypothetical `target_cap`.
The solution implemented approaches what we would have with a `tcg_gen_cap_ld/st` interface.

Moreover, receiving the full capability register information gives us the capability fields and tag to store within the `log_meminfo_t` in the log info, without having to worry about the ordering of the write/reads.
If this was not the case, it would be hard to identify a set of writes as a capability+tag from the logging backend alone. This is relatively harmless for text tracing, but we need to reliably get the capability value if we want to support future uses of binary formats or other filtering that need to reason about the capability being written.
Note that this becomes an harder problem in presence of multiple writes, like aarch64 load/store pair.